flink-python/src/main/java/org/apache/flink/python/env/AbstractPythonEnvironmentManager.java

```diff
@@ -166,6 +166,14 @@ public Map<String, String> constructEnvironmentVariables(String baseDirectory)
 
         constructFilesDirectory(env, baseDirectory);
 
+        if (this.dependencyInfo.getPythonPath() != null) {
+            List<String> configuredPythonPath = new ArrayList<>();
+            configuredPythonPath.add(this.dependencyInfo.getPythonPath());
+            appendToPythonPath(env, configuredPythonPath);
```

Comment:
IS the PYTHONPATH used elsewhere? Will this affect any other path resolution on the TM?

Reply:
Ideally it should not affect path resolution on TM.